xarray/core/formatting.py

```diff
@@ -408,10 +408,13 @@ def coords_repr(coords, col_width=None, max_rows=None):
     )
 
 
-def summarize_index(
-    name: Hashable, index, col_width: int, max_width: int = None, is_index: bool = False
-):
-    return pretty_print(f"    {name} ", col_width) + f"{repr(index)}"
+def summarize_index(name: Hashable, index, col_width: int):
+    if hasattr(index, "_repr_inline_"):
+        repr_ = index._repr_inline_()
+    else:
+        repr_ = repr(index)
```

Comment:
Alternatively we could provide a default implementation for `Index._repr_inline_()` that only shows the class name. What do you think?

Defining `_repr_inline_` in the base class would be it a bit more discoverable for anyone who wants to create an `Index` subclass. Also, `repr(index)` may not be displayed nicely on one line (for the html repr the full index repr could be displayed using the same button icon than for the variables).

Reply:
Should we move the inline logic into an `inline_index_repr` function instead? Like `inline_variable_array_repr` vs. `summarize_variable` and so we can reuse it in the html repr.

Reply:
oh, yes, that makes sense. I'm looking into the HTML repr at the moment and actually redefined that function there...